tests/test_dataclasslike.py

```diff
@@ -0,0 +1,68 @@
+import types
+import unittest
+from dataclasses import dataclass, is_dataclass
+from unittest import skipUnless
+
+from useful_types.experimental import DataclassLike
+
+class DataclassLikeTests(unittest.TestCase):
+    def test_basics(self):
+        # As an abstract base class for all dataclasses,
+        # it makes sense for DataclassLike to also be considered a dataclass
```

Comment:
I see, the runtime version does have the `@dataclass` decorator, but the `if TYPE_CHECKING` version doesn't, and that's the one I was looking at.

I'm talking about the behavior of `is_dataclass()` here, not `issubclass()`, so most of the examples you cite don't really apply. `collections.abc.Callable` really is callable.

Reply:
I think it's more like these:

```
>>> inspect.isfunction(types.FunctionType)
False
>>> inspect.iscode(types.CodeType)
False
>>> inspect.ismodule(types.ModuleType)
False
```

Reply:
Just looking at this, sorry, I'm confused. Which version of `DataclassLike` has the `@dataclass` decorator?

Since I'm not sure I understand the situation, I don't have an opinion yet. We currently disallow subclassing, which is wise, if we also make the protocol not runtime checkable then there aren't really any runtime use cases for this, which could render a decision here somewhat academic.

Reply:
Currently no version of DataclassLike has the @dataclass decorator, because I addressed Jelle's other comment elsewhere where he said he wanted the runtime implementation and the static implementation to be as similar as possible :) An earlier version had the decorator on the runtime implementation, but not the static implementation.

`dataclasses.is_dataclass(DataclassLike)` still returns `True`, however, because all the function looks for is the presence of a `__dataclass_fields__` class variable, which DataclassLike has.

> if we also make the protocol not runtime checkable then there aren't really any runtime use cases for this, which could render a decision here somewhat academic.

That's a good point actually — maybe we should just do that, on the grounds that `dataclasses` already has ways to determine whether or not something is a dataclass at runtime (the `is_dataclass` function)

Reply:
The PR no longer proposes making it runtime-checkable!